ios/FluentUI/DrawerVnext/Drawer.swift

```diff
@@ -0,0 +1,266 @@
+protocol DrawerStateDelegate: AnyObject {
+    @objc optional func drawerDidExpand()
+    @objc optional func drawerDidCollapsed()
+}
+
+public class DrawerTokens: ObservableObject {
+
+    @Published public var shadowColor: Color!
+    @Published public var shadowOpacity: Double!
+    @Published public var shadowBlur: CGFloat!
+    @Published public var shadowDepth: [CGFloat]! // x,y axis
+
+    public init() {
+        self.themeAware = true
+        didChangeAppearanceProxy()
+    }
+
+    @objc open func didChangeAppearanceProxy() {
+        let appearanceProxy = StylesheetManager.S.DrawerTokens
+        shadowColor = Color(appearanceProxy.shadowColor)
+        shadowOpacity = Double(appearanceProxy.shadowOpacity)
+        shadowBlur = appearanceProxy.shadowBlur
+        shadowDepth = appearanceProxy.shadowDepth
+    }
+}
+
+// Mark - Drawer View
+
+@objc public enum DrawerDirection: Int, CaseIterable {
+    case left
+    case right
+}
+
+/**
+ `Drawer` is used to present a content that is currently in hideout mode when collapsed and when expanded is used to present content view partially either on the left or right.
+ 
+ Currently `Drawer` only support horizontal axis and is expanded by default from left side of the screen unless explicitly specified
+ 
+ Set `Content` to provide content for the drawer.
+ */
+public struct Drawer<Content: View>: View {
+
+    @ObservedObject var state: DrawerState
+    @ObservedObject var tokens: DrawerTokens
+
+    private let backgroundLayerColor: Color = .black
+    private var backgroundLayerOpacity: Double {
+        return state.backgroundDimmed ? 0.5 : 0
+    }
+    private let backgroundLayerAnimation: Animation = .default
+
+    private let percentWidthOfContent: CGFloat = 0.9
+    private let percentSnapWidthOfScreen: CGFloat = 0.225
+
+    // content view on top of
+    public var content: Content
+
+    // `ivar` to keep track of dragged offset
+    @State private var draggedOffsetWidth: CGFloat?
+    private func computedOffset(screenSize: CGSize) -> CGFloat {
+
+        if let draggedOffsetWidth = draggedOffsetWidth {
+            return draggedOffsetWidth
+        }
+
+        var width = CGFloat.zero
+        let contentWidth = screenSize.width * percentWidthOfContent
+        if !state.isExpanded { // if closed then hidden behind the screen
+            width = state.presentationDirection == .left ? -contentWidth : contentWidth
+        }
+
+        return width
+    }
+
+    public init(content: Content) {
+        self.content = content
+        tokens = DrawerTokens()
+        state = DrawerState()
+    }
+
+    public var body: some View {
+        GeometryReader { reader in
+            HStack {
+                if state.presentationDirection == .right {
+                    Spacer()
+                }
+                content
+                    .frame(width: reader.size.width * percentWidthOfContent)
```

Comment:
The current drawer does not change its width in horizontal orientation.
This seems to be the right behavior for the left nav at least.

![drawer_horizontal](https://user-images.githubusercontent.com/68076145/102935099-4f1d5580-445a-11eb-86b9-b1afa3929cfc.png)
![drawer_vnext_horizontal](https://user-images.githubusercontent.com/68076145/102935103-517faf80-445a-11eb-8b1c-3697c8235fb1.png)

Reply:
Thanks for catching this, I have updated to match the default functionality. [I've added a small hack to override the default behavior] 
I believe the approach should be to either use `preferredContentSize` from content view controller or stretch to full screen because if any app needs the functionality other way around then currently there is no way to do so. 

We can revisit this when adding vertical support for drawer controller.